app/helpers/application_helper/button/generic_show_list_feature_button.rb

```diff
@@ -0,0 +1,18 @@
+class ApplicationHelper::Button::GenericShowListFeatureButton < ApplicationHelper::Button::ButtonNewDiscover
```

Comment:
The button should not be inherited from `ButtonNewDiscover` because it has nothing to do with that. Also it's not really generic if it accepts a `managers` attribute. Let's name it `PolymorphicConditionalButton` and inherit it from `GenericFeatureButton`, so you can spare the `@feature` part from the initialize both here and in the specific one. 

The `target_class` attribute should be maybe `parent_class` and the `managers` could be `child_classes`, this way it would be truly generic. I would also avoid using the `managers` as an instance variable and setting it as a param, as it's not that declarative. You could define it as a method like this to have a default value:
```
def child_classes
  @parent_class.descendants
end
```

Then in the class above you could just redefine the method.

Reply:
Okay, the inheritance might not work :see_no_evil: sorry, but the rest still is valid.

Reply:
I did the modifications. is it ok ?